cmd/release-notes/main.go

```diff
@@ -54,154 +55,164 @@ type options struct {
+	RunE:          run,
+}
+
+func init() {
 	// githubToken contains a personal GitHub access token. This is used to
 	// scrape the commits of the Kubernetes repo.
-	flags.StringVar(
-		&o.githubToken,
+	cmd.PersistentFlags().StringVar(
+		&opts.githubToken,
 		"github-token",
 		envDefault("GITHUB_TOKEN", ""),
 		"A personal GitHub access token (required)",
 	)
 
 	// githubOrg contains name of github organization that holds the repo to scrape.
-	flags.StringVar(
-		&o.githubOrg,
+	cmd.PersistentFlags().StringVar(
+		&opts.githubOrg,
 		"github-org",
 		envDefault("GITHUB_ORG", "kubernetes"),
 		"Name of github organization",
 	)
 
 	// githubRepo contains name of github repository to scrape.
-	flags.StringVar(
-		&o.githubRepo,
+	cmd.PersistentFlags().StringVar(
+		&opts.githubRepo,
 		"github-repo",
 		envDefault("GITHUB_REPO", "kubernetes"),
 		"Name of github repository",
 	)
 
 	// output contains the path on the filesystem to where the resultant
 	// release notes should be printed.
-	flags.StringVar(
-		&o.output,
+	cmd.PersistentFlags().StringVar(
+		&opts.output,
 		"output",
 		envDefault("OUTPUT", ""),
 		"The path to the where the release notes will be printed",
 	)
 
 	// branch is which branch to scrape.
-	flags.StringVar(
-		&o.branch,
+	cmd.PersistentFlags().StringVar(
+		&opts.branch,
 		"branch",
 		envDefault("BRANCH", "master"),
 		"Select which branch to scrape. Defaults to `master`",
 	)
 
 	// startSHA contains the commit SHA where the release note generation
 	// begins.
-	flags.StringVar(
-		&o.startSHA,
+	cmd.PersistentFlags().StringVar(
+		&opts.startSHA,
 		"start-sha",
 		envDefault("START_SHA", ""),
 		"The commit hash to start at",
 	)
 
 	// endSHA contains the commit SHA where the release note generation ends.
-	flags.StringVar(
-		&o.endSHA,
+	cmd.PersistentFlags().StringVar(
+		&opts.endSHA,
 		"end-sha",
 		envDefault("END_SHA", ""),
 		"The commit hash to end at",
 	)
 
 	// startRev contains any valid git object where the release note generation
 	// begins. Can be used as alternative to start-sha.
-	flags.StringVar(
-		&o.startRev,
+	cmd.PersistentFlags().StringVar(
+		&opts.startRev,
 		"start-rev",
 		envDefault("START_REV", ""),
 		"The git revision to start at. Can be used as alternative to start-sha.",
 	)
 
 	// endRev contains any valid git object where the release note generation
 	// ends. Can be used as alternative to start-sha.
-	flags.StringVar(
-		&o.endRev,
+	cmd.PersistentFlags().StringVar(
+		&opts.endRev,
 		"end-rev",
 		envDefault("END_REV", ""),
 		"The git revision to end at. Can be used as alternative to end-sha.",
 	)
 
 	// repoPath contains the path to a local Kubernetes repository to avoid the
 	// delay during git clone
-	flags.StringVar(
-		&o.repoPath,
+	cmd.PersistentFlags().StringVar(
+		&opts.repoPath,
 		"repo-path",
-		envDefault("REPO_PATH", ""),
-		"Path to a the local Kubernetes repository",
+		envDefault("REPO_PATH", filepath.Join(os.TempDir(), "k8s-repo")),
```

Comment:
@hoegaarden referring to https://github.com/kubernetes/release/pull/969#pullrequestreview-331812410

I changed the default repo to be `/tmp/k8s-repo`. Can you remove that directory and test it again on your side? For me it works on linux. Are you on macOS?

Reply:
Ok ... clearing that worked.
Stupid me was too fast with the `rm`, would be interesting to know what went wrong there, what the state & content of that repo was.

Reply:
The thing is that the state is different (but not wrong right now) on clone and reopen:
- On clone you only clone the repo like every other git clone would do it
- On refresh we fetch all (!) references to be available locally

I think we have to fix that, to do a refresh after the clone in any case... :thinking:

Reply:
What do we need the local clone anyway? The parts I have seen ask the GtiHub API for all the things.

On the long run, I think we should strive for either doing everything with a local repo only and not talking to GitHub (except of course making sure the repo is up to date). Not sure if this is even possible, would we have all the PR body in the commit message of the merge commits? Or use the API only and don't depend on a local clone.